File: flink-jobs/src/main/java/flinkfintechpoc/jobs/WordCountJob.java
package flinkfintechpoc.jobs;
import flinkfintechpoc.Tokenizer;
import org.apache.flink.api.common.typeinfo.TypeInformation;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

import java.util.Arrays;
import java.util.List;

public class WordCountJob {

  public static void main(String[] args) throws Exception {
    final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

    List<String> lines = Arrays.asList(
        "Apache Flink is a powerful stream processor",
        "Flink is great for both batch and stream",
        "Stream processing with Flink is efficient"
    );

    DataStream<String> text = env.fromCollection(lines, TypeInformation.of(String.class));

    DataStream<Tuple2<String, Integer>> counts = text
        .flatMap(new Tokenizer())
        .keyBy(tuple -> tuple.f0)
        .sum(1);

    counts.print();
    env.execute("Flink 2.0 WordCount with Static Data");
  }
}

File: flink-jobs-python/src/jobs/word_count_job.py
#!/usr/bin/env python3
"""
Flink Word Count Job in Python
Equivalent to the Java WordCountJob
"""

from pyflink.datastream import StreamExecutionEnvironment
from pyflink.table import StreamTableEnvironment
from pyflink.common import Types
from pyflink.datastream.functions import FlatMapFunction, MapFunction


class Tokenizer(FlatMapFunction):
    """Tokenizer function to split text into words"""
    
    def flat_map(self, value):
        words = value.lower().split()
        for word in words:
            # Filter out empty strings and punctuation
            cleaned_word = ''.join(c for c in word if c.isalnum())
            if cleaned_word:
                yield (cleaned_word, 1)


def main():
    """Main function to execute the Word Count job"""
    
    # Create streaming execution environment
    env = StreamExecutionEnvironment.get_execution_environment()
    env.set_parallelism(1)
    
    # Sample data
    lines = [
        "Apache Flink is a powerful stream processor",
        "Flink is great for both batch and stream", 
        "Stream processing with Flink is efficient"
    ]
    
    # Create data stream from collection
    text_stream = env.from_collection(lines, type_info=Types.STRING())
    
    # Apply word count transformation
    word_counts = text_stream \
        .flat_map(Tokenizer(), output_type=Types.TUPLE([Types.STRING(), Types.INT()])) \
        .key_by(lambda x: x[0]) \
        .sum(1)
    
    # Print results
    word_counts.print()
    
    # Execute the job
    env.execute("Python Flink WordCount Job")


if __name__ == "__main__":
    main()
